gtsam/discrete/DiscreteFactor.h

```diff
@@ -107,12 +126,35 @@ class GTSAM_EXPORT DiscreteFactor : public Factor {
   /// Compute error for each assignment and return as a tree
   virtual AlgebraicDecisionTree<Key> errorTree() const;
 
-  /// Multiply in a DecisionTreeFactor and return the result as
-  /// DecisionTreeFactor
-  virtual DecisionTreeFactor operator*(const DecisionTreeFactor&) const = 0;
+  /// Multiply in a DiscreteFactor and return the result as
+  /// DiscreteFactor
+  virtual DiscreteFactor::shared_ptr operator*(
```

Comment:
Should this be a `DiscreteFactor::shared_ptr` or a `DiscreteFactor&` (aka a reference?)